docs/sphinx_doc/en/source/tutorial/206-prompt.md

```diff
@@ -1,16 +1,279 @@
+   `role` field only supports `"user"` and `"assistant"`.
+
+2. Some model APIs require `"user"` and `"assistant"` must speak alternatively,
+   and `"user"` must speak in the beginning and end of the input messages list.
+   Such requirements make it difficult to build a multi-agent conversation
+   when the agent may act as many different roles and speak continuously.
+
+To help beginners to quickly start with AgentScope, we provide the
+following built-in strategies for most chat and generation related model APIs.
+
+## Built-in Prompt Strategies
+
+In AgentScope, we provide built-in strategies for the following chat and
+generation model APIs.
+
+- [`OpenAIChatWrapper`](#openaichatwrapper)
+- [`DashScopeChatWrapper`](#dashscopechatwrapper)
+- [`OllamaChatWrapper`](#ollamachatwrapper)
+- [`OllamaGenerationWrapper`](ollamagenerationwrapper)
+- [`GeminiChatWrapper`](#geminiwrapper)
+
+These strategies are implemented in the `format` functions of the model
+wrapper classes.
+It accepts `Msg` objects, a list of `Msg` objects, or their mixture as input.
+
+### `OpenAIChatWrapper`
+
+`OpenAIChatWrapper` encapsulates the OpenAI chat API, it takes a list of
+dictionaries as input, where the dictionary must obey the following rules
+(updated in 2024/03/22):
+
+- Require `role` and `content` fields, and an optional `name` field.
+- The `role` field must be either `"system"`, `"user"`, or `"assistant"`.
+
+#### Prompt Strategy
+
+In OpenAI Chat API, the `name` field enables the model to distinguish
+different speakers in the conversation. Therefore, the strategy of `format`
+function in `OpenAIChatWrapper` is simple:
+
+- `Msg`: Pass a dictionary with `role`, `content`, and `name` fields directly.
+- `List`: Parse each element in the list according to the above rules.
+
+An example is shown below:
+
+```python
+from agentscope.models import OpenAIChatWrapper
+from agentscope.message import Msg
+
+model = OpenAIChatWrapper(
+    config_name="", # empty since we directly initialize the model wrapper
+    model_name="gpt-4",
+)
+
+prompt = model.format(
+   Msg("system", "You're a helpful assistant", role="system"),   # Msg object
+   [                                                             # a list of Msg objects
+      Msg(name="Bob", content="Hi.", role="assistant"),
+      Msg(name="Alice", content="Nice to meet you!", role="assistant"),
+   ],
+)
+print(prompt)
+```
+
+```bash
+[
+  {"role": "system", "name": "system", "content": "You are a helpful assistant"},
+  {"role": "assistant", "name": "Bob", "content": "Hi."},
+  {"role": "assistant", "name": "Alice", "content": "Nice to meet you!"),
+]
+```
+
+### `DashScopeChatWrapper`
+
+`DashScopeChatWrapper` encapsulates the DashScope chat API, which takes a list of messages as input. The message must obey the following rules (updated in 2024/03/22):
+
+- Require `role` and `content` fields, and `role` must be either `"user"`
+  or `"assistant"`.
```

Comment:
Use of the `system` role is optional and, if present, must be at the first message of the list.

Reply:
Add rule